ompi/mca/osc/ucx/osc_ucx_comm.c

```diff
@@ -353,20 +353,20 @@ static inline int get_dynamic_win_info(uint64_t remote_addr, ompi_osc_ucx_module
     assert(win_count > 0 && win_count <= OMPI_OSC_UCX_ATTACH_MAX);
 
     temp_dynamic_wins = (ompi_osc_dynamic_win_info_t *)(temp_buf + sizeof(uint64_t));
-    contain = ompi_osc_find_attached_region_position(temp_dynamic_wins, 0, win_count,
+    *win_idx = ompi_osc_find_attached_region_position(temp_dynamic_wins, 0, win_count,
                                                      remote_addr, 1, &insert);
-    assert(contain >= 0 && (uint64_t)contain < win_count);
+    assert(*win_idx >= 0 && (uint64_t)*win_idx < win_count);
```

Comment:
We should propagate this as an error and return `MPI_ERR_RMA_RANGE` to the user

Reply:
Yeah. asserts in production code for non-fatal errors are not ok!

Reply:
Note also that assert() is a no-op in an optimized build.

Reply:
fixed, forced update